teps/0101-decouple-catalog-organization-and-reference.md

```diff
@@ -0,0 +1,233 @@
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [Implementation Pull request(s)](#implementation-pull-request-s)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+As described in [TEP-0003](./0003-tekton-catalog-organization.md) :
+The Tekton catalog is a collection of blessed Tekton resources that
+can be used with any system that supports the Tekton API. We also have
+the [hub](https://hub.tekton.dev) to be able to search for Tasks.
+
+This proposes to make the *Hub* and any other **official** mean of
+consuming Tekton resources from the community decoupled from where we
+author them. In a gist, this aims to make the repository
+`tektoncd/catalog` an implementation detail, and only support
+long-term, non-changing URI to refer to tasks (Those URIs could be
+`https://` endpoint such as the Hub or docker images, …).
+
+## Motivation
+
+As of today, users of the tekton task are directly refering to the
+GitHub URL to get Task from the catalog. This creates a hard coupling
+between the way we organize tasks inside Tekton and the way user
+consume it. This is forcing us into one possible catalog organization
+and prevent us to make drastic changes to this organization without
+breaking users.
+
+This enhancement aim to decouple the catalog organization from users
+consumption of those resources. This enhancement proposal is mainly
+touching the catalog and the hub components. This would also push the
+hub to be the main entrypoint for users to search, list and get tekton
+resources.
+
+### Goals
+
+- Allowing for the maintainers of `tektoncd/catalog` to change the
+  organization without affecting the consumption of the resources.
+- Define a set of standard, supported ways to get a tekton resource
+  (`Task`, `Pipeline`, …)
+
+### Non-Goals
+
+- Modify the current organization of the `tektoncd/catalog` repository.
+
+### Use Cases
+
+<!--
+Describe the concrete improvement specific groups of users will see if the
+Motivations in this doc result in a fix or feature.
+
+Consider both the user's role (are they a Task author? Catalog Task user?
+Cluster Admin? etc...) and experience (what workflows or actions are enhanced
+if this problem is solved?).
+-->
+
+## Requirements
+
+- A end-user should never refer / install task from a URL that points
+  directly to the github repository (like https://raw.githubusercontent.com/tektoncd/catalog/main/task/git-clone/0.5/git-clone.yaml).
+
+## Proposal
+
+This is a very very early draft, this will need to be better written and detailed, right now it’s just a “list” of things
+
+- User fetch tekton catalog resources throught the following “medium”
+  - oci image reference(s) : gcr.io/tekton-releases/catalog/upstream/task/{name}:{version}
```

Comment:
The problem with OCI references is that they're bound to a specific container registry... so if tomorrow we want to switch to icr or ghcr or quay we have an issue. 
I wonder if it would be possible at all to have a cr.tekton.dev that proxies to whatever CR we use at the moment.